src/IssueViz.Security/Hotspots/LocalHotspotsStore.cs

```diff
@@ -0,0 +1,214 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Security.IssuesStore;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.Hotspots
+{
+    /// <summary>
+    /// Methods for modification of <see cref="ILocalHotspotsStore"/>
+    /// </summary>
+    public interface ILocalHotspotsStoreUpdater
+    {
+        void AddOrUpdate(string filePath, IEnumerable<IAnalysisIssueVisualization> hotspots);
```

Comment:
It's not clear from the method signature whether the add/update is happening at file-level or hotspot-level e.g. if I make the following two calls:

```
  updater.AddOrUpdate("file1.txt", new [] { hotspot1 });
  updater.AddOrUpdate("file1.txt", new [] { hotspot2 });
```

Does the second call completely replace the list of hotspots for the file, or does it add to it?

Reply:
Maybe rename to UpdateHotspotsForFile?

Reply:
Yep, that's better.